api/tests/profile/integration/scripts/sixth-grade-attestation-reward_test.js

```diff
@@ -0,0 +1,319 @@
+import sinon from 'sinon';
+
+import { CampaignParticipationStatuses } from '../../../../src/prescription/shared/domain/constants.js';
+import {
+  fetchUserIds,
+  PRODUCTION_SIXTH_GRADE_TARGET_PROFILE_IDS,
+  SixthGradeAttestationRewardScript,
+} from '../../../../src/profile/scripts/sixth-grade-attestation-reward.js';
+import { catchErr, databaseBuilder, expect } from '../../../test-helper.js';
+
+describe('Integration | Profile | Scripts | sixth-grade-attestation-reward', function () {
+  describe('options', function () {
+    it('parses dates correctly', function () {
+      const startDate = '2024-01-01';
+      const endDate = { whut: 'idontknow' };
+      const script = new SixthGradeAttestationRewardScript();
+      const { options } = script.metaInfo;
+      const parsedDate = options.start.coerce(startDate);
+      expect(parsedDate).to.be.a.instanceOf(Date);
+      expect(() => options.end.coerce(endDate)).to.throw();
+    });
+
+    it('parses limit correctly', function () {
+      const script = new SixthGradeAttestationRewardScript();
+      const { options } = script.metaInfo;
+      const parsedLimit = options.limit.coerce('12');
+      expect(parsedLimit).to.equal(12);
+    });
+  });
+
+  describe('#fetchUsers', function () {
+    it('should not return the user if the participation date is not included between the start date and the end date ', async function () {
+      const { id: targetProfileId } = databaseBuilder.factory.buildTargetProfile({
+        id: PRODUCTION_SIXTH_GRADE_TARGET_PROFILE_IDS[0],
+      });
+      const campaign = databaseBuilder.factory.buildCampaign({ targetProfileId });
+      const { userId } = databaseBuilder.factory.buildCampaignParticipation({
+        campaignId: campaign.id,
+        status: CampaignParticipationStatuses.SHARED,
+        createdAt: '2024-12-02',
+      });
+      databaseBuilder.factory.buildCampaignParticipation({
+        campaignId: campaign.id,
+        status: CampaignParticipationStatuses.SHARED,
+        createdAt: '2024-12-07',
+      });
+      databaseBuilder.factory.buildCampaignParticipation({
+        campaignId: campaign.id,
+        status: CampaignParticipationStatuses.SHARED,
+        createdAt: '2024-11-22',
+      });
+
+      await databaseBuilder.commit();
+      const userIds = await fetchUserIds(new Date('2024-12-01'), new Date('2024-12-03'));
+      expect(userIds).to.have.lengthOf(1);
+      expect(userIds).to.contains(userId);
+    });
+
+    it('should not return the user if the participation status is different from started', async function () {
```

Comment:
**thought**: le it n'est pas bon. Ca serait plutôt `it should not return the user if the participation status is started`